src/org/scada_lts/web/mvc/api/ReportsAPI.java

```diff
@@ -0,0 +1,309 @@
+import com.serotonin.mango.vo.report.ReportVO;
+import com.serotonin.mango.web.dwr.beans.DataPointBean;
+import com.serotonin.mango.web.dwr.beans.RecipientListEntryBean;
+import com.serotonin.mango.web.email.MangoEmailContent;
+import com.serotonin.web.dwr.DwrResponseI18n;
+import com.serotonin.web.i18n.I18NUtils;
+import com.serotonin.web.i18n.LocalizableMessage;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.scada_lts.dao.report.ReportInstancePointDAO;
+import org.scada_lts.mango.service.DataPointService;
+import org.scada_lts.mango.service.EventService;
+import org.scada_lts.mango.service.ReportService;
+import org.scada_lts.mango.service.SystemSettingsService;
+import org.scada_lts.utils.SQLPageWithTotal;
+import org.scada_lts.web.mvc.api.dto.EventCommentDTO;
+import org.scada_lts.web.mvc.api.dto.EventDTO;
+import org.scada_lts.web.mvc.api.json.JsonEventSearch;
+import org.scada_lts.web.mvc.api.json.JsonIdSelection;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+import java.util.*;
+
+/**
+ * Simple controller for Reports in Scada-LTS
+ *
+ * @author Sergio Selvaggi <sselvaggi@softq.pl>
+ */
+@Controller
+@RequestMapping(value = "/api/reports")
+public class ReportsAPI {
+
+    private static final Log LOG = LogFactory.getLog(ReportsAPI.class);
+    private DataPointService dataPointService = new DataPointService();
+
+    @Resource
+    private ReportService reportService;
+
+    @Resource
+    private SystemSettingsService systemSettingsService;
+
+    /**
+     * Get Reports related to user
+     *
+     * @param request     HTTP request with user data
+     * @return ReportVO List
+     */
+    @PostMapping(value = "/search")
+    public ResponseEntity<List<ReportVO>> getAll(@RequestParam Map<String, String> query, HttpServletRequest request) {
+        LOG.info("GET::/api/reports/getAll");
+        try {
+            User user = Common.getUser(request);
+            if (user != null) {
+                return new ResponseEntity<List<ReportVO>>(
+                    reportService.search(query),
+                    HttpStatus.OK
+                );
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+
+
+    /**
+     * Save Reports related to user
+     *
+     * @param request     HTTP request with user data
+     * @return ReportVO List
+     */
+    @PostMapping(value = "/save")
+    public HttpEntity<String> save(@RequestBody Map<String, Object> query, HttpServletRequest request) {
```

Comment:
Consider creation of ReportDTO.java class that you will be receiving from the User request in a body. It should match the report object that are you receiving from user.